compiler/src/dotty/tools/dotc/ast/tpd.scala

```diff
@@ -1149,6 +1149,10 @@ object tpd extends Trees.Instance[Type] with TypedTreeInfo {
         var allAlts = denot.alternatives
           .map(denot => TermRef(receiver.tpe, denot.symbol))
           .filter(tr => typeParamCount(tr) == targs.length)
+          .filter { _.widen match {
+            case MethodTpe(_, _, x: MethodType) => !x.isImplicitMethod
```

Comment:
What about curried methods that have implicit parameters as the 3rd parameter clause? Should we accept them? I guess there's no problem with those since we would not have to resolve them right away, right?

Reply:
There can be a problem... It looks like `applyOverloaded` resolves a method name looking only at the first argument group. E.g.:

```scala
def f(x: Int)(y: Int) given Z: Int
def f(x: String)(y: Int): Int
applyOverloaded("f", x: Any)
```

In both cases, the logic above infers `x.isImplicitMethod` to be `false`, so it cannot differentiate and can select the one with the implicit args at the end. Which it will never be able to call if we are after Typer.

Reply:
Ah, OK. So we need to exclude all implicit methods, even of the implicit comes later.

Reply:
@odersky do you think this is no longer a problem?